src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2101,6 +2101,10 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 dest_type = ASR::down_cast<ASR::ttype_t>(
                         ASR::make_Character_t(al, loc, right_type2->m_kind,
                         dest_len, nullptr));
+            } else if ((left_is_int && ASRUtils::expr_value(left) == nullptr) 
+                || (right_is_int && ASRUtils::expr_value(right) == nullptr)) {
+                dest_type = ASRUtils::TYPE(ASR::make_Character_t(al,
+                            loc, 1, 1, nullptr));
```

Comment:
```suggestion
                dest_type = ASRUtils::TYPE(ASR::make_Character_t(al,
                            loc, 1, -1, nullptr));
```

Reply:
@Shaikh-Ubaid I am curious, why does setting the length to -1 not cause an issue?

Reply:
Setting `length to 1` could potentially fail in future (for example when there are operations that involve compile-time length). Therefore, it is better to use the appropriate length value in the generated ASR.

`length >= 0` indicates a compile-time length
`length = -1` indicates length is not available and need to be computed at runtime (For example `char*` in c)
`length = -2` indicates length is not available, but the variable is allocatable 
`length = -3` indicates length is available in the form of an expression given as the last parameter of `make_Character_t`.

You can also check the following for more details.

https://github.com/lcompilers/lpython/blob/3ccdfdb6d6da234a31a0b28fb627691cc9d7e214/src/libasr/codegen/asr_to_fortran.cpp#L174-L187

Reply:
:star_struck: Thanks a lot for this information @Shaikh-Ubaid!